File: 02 - api/javaScrpit/fipe/src/controller/fipe_controller.ts
import { Request, Response } from "express";
import FipeService from "../service/fipe_service";

const fipeService = new FipeService();

export const consultarMarcas = async (req: Request, res: Response) => {
  const data = await fipeService.consultarMarcas();
  res.json(data);
};

export const consultarModelos = async (req: Request, res: Response) => {
  const marca = parseInt(req.params.marca);
  const data = await fipeService.consultarModelos(marca);
  res.json(data);
};

export const consultarAnos = async (req: Request, res: Response) => {
  const marca = parseInt(req.params.marca);
  const modelo = parseInt(req.params.modelo);
  const data = await fipeService.consultarAnos(marca, modelo);
  res.json(data);
};

export const consultarValor = async (req: Request, res: Response) => {
  const marca = parseInt(req.params.marca);
  const modelo = parseInt(req.params.modelo);
  const ano = req.params.ano;
  const data = await fipeService.consultarValor(marca, modelo, ano);
  res.json(data);
};

File: 022024/02 - api/python/api/fipe/controller/fipe_controller.py
# fipe/controllers/fipe_controller.py
from ninja import NinjaAPI
from fipe.service.fipe_service import FipeService

api = NinjaAPI()
fipe_service = FipeService()

@api.get("/marcas")
def consultar_marcas(request):
    return fipe_service.consultar_marcas()

@api.get("/modelos/{marca}")
def consultar_modelos(request, marca: int):
    return fipe_service.consultar_modelos(marca)

@api.get("/anos/{marca}/{modelo}")
def consultar_anos(request, marca: int, modelo: int):
    return fipe_service.consultar_anos(marca, modelo)

@api.get("/valor/{marca}/{modelo}/{ano}")
def consultar_valor(request, marca: int, modelo: int, ano: str):
    return fipe_service.consultar_valor(marca, modelo, ano)